phil/cnx-routes.md

```diff
@@ -0,0 +1,302 @@
+
+1. Browser requests [https://cnx.org/contents/{UUID}](https://cnx.org/contents/031da8d3-b525-429c-80cf-6c8ed997733a)
+1. JavaScript requests [https://archive.cnx.org/contents/{UUID}](https://archive.cnx.org/contents/031da8d3-b525-429c-80cf-6c8ed997733a)
+1. server redirects to [https://archive.cnx.org/contents/{UUID}@{VERSION}](https://archive.cnx.org/contents/031da8d3-b525-429c-80cf-6c8ed997733a@9.74)
+1. AJAX response contains the short UUID for the Book (`Ax2o07Ul`) and the Preface Page (`HR_VN3f7@3`)
+1. JavaScript requests the Preface Page with the book ID to get the correct context
+1. JavaScript changes the browser URL to [https://cnx.org/contents/{BOOK_SHORT_ID}@{VER}:{PAGE_SHORT_ID}@{VER}}/{TITLE_SLUG}](https://cnx.org/contents/Ax2o07Ul@9.74:HR_VN3f7@3/Introduction-to-Science-and-th)
+
+
+## Publishing Content
+
+Publishing content requires POSTing a specially-formatted ePUB as a payload.
+
+This payload contains all the Books (usually 1), Pages, Resources, and metadata (like permissions).
+
+See the [^cnx-publishing-api] for more details on the format and URLs.
+
+
+
+
+[^cnx-archive]: https://github.com/Connexions/cnx-archive
+[^cnx-archive-search-api]: https://github.com/Connexions/cnx-archive/blob/master/docs/search_api_doc.rst
+[^cnx-publishing]: https://github.com/Connexions/cnx-publishing
+[^cnx-publishing-api]: https://github.com/Connexions/cnx-publishing#http-api
+[^webview]: https://github.com/Connexions/webview
+
+
+---
+---
+---
+---
+---
+---
+---
+---
+---
+
+Below be dragons
+
+---
+---
+---
+---
+---
+---
+---
+---
+---
+---
+---
+---
+---
+
+
+All the routes in https://cnx.org, https://archive.cnx.org, and CNX Publishing
+
+# CNX Webview
+
+- `/`
+- `/workspace`
+- `/contents`
+- `/browse`
+- `/tos`
+- `/license`
+- `/users/role-acceptance/{details}/`
+- `/contents/`
+- `/contents/{contentPattern}'`
+```
+  # Match and extract uuid and page numbers separated by a colon
+  contentPattern =
+    ///
+    ([^:@/]+)     # uuid up to delimiter
+    @?            # Optional @
+    ([^:/?]*)     # Optional Revision
+    :?            # Optional ':'
+    ([^/?]*)      # Optional Page number or uuid
+    /?            # Optional '/'
+    ([^?]*)       # Optional Segment of title
+    (\?.*)?       # Optional params
+    ///
+```
+
+
+- `/donate/{/segment1?}{/segment2?}{/segment3?}{/segment4?}`
+- `/search{?q}`
+- `/about{?}`
+
+
+# CNX Archive
+
+- `/contents/{ident_hash}{:page_ident_hash?}{/ignore?}.html`
+- `/contents/{ident_hash}{:page_ident_hash?}{/ignore?}.json`
+- `/contents/{ident_hash}{:page_ident_hash?}{/ignore?}`
+- `/resources/{hash}{/ignore?}`
+- `/exports/{ident_hash}.{type}{/ignore?}`
+- `/extras/{ident_hash}`
+- `/search`
+- `/search/{ident_hash}`
+- `/search/{ident_hash}{:page_ident_hash?}`
+- `/extras`
+- `/sitemap.xml`
+- `/robots.txt`
+- `/content/{objid}{/ignore?}`
+- `/content/{objid}/latest{/ignore?}{/filename?}`
+- `/content/{objid}/{objver}{/ignore?}{/filename?}`
+
```

Comment:
You can get a more verbose format of these routes, their view code location and the methods they accept by running `proutes <ini_config_file>`. At the very least it might be worth noting that you can get the most updated list of routes using this utility.